server/src/test/java/com/google/coffeehouse/servlets/GetProfileServletTest.java

```diff
@@ -0,0 +1,145 @@
+
+package com.google.coffeehouse.servlets;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.coffeehouse.common.Person;
+import com.google.coffeehouse.storagehandler.StorageHandlerApi;
+import com.google.coffeehouse.storagehandler.StorageHandler;
+import com.google.gson.Gson;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+/**
+ * Unit tests for {@link GetProfileServlet}.
+ */
+public class GetProfileServletTest {
+  private static final String USER_ID = "predetermined-identification-string";
+  private static final String EMAIL = "email@test.com";
+  private static final String NICKNAME = "test";
+  private static final String PRONOUNS = "they";
+  private static final String JSON = String.join("\n",
+      "{",
+      "  \"" + Person.USER_ID_FIELD_NAME + "\" : \"" + USER_ID + "\"",
+      "}");
+  private static final String NO_USER_ID_JSON = "{}";
+  private static final String PROFILE_NOT_FOUND_JSON = String.join("\n",
+      "{",
+      "  \"" + Person.USER_ID_FIELD_NAME + "\" : \"\"",
+      "}");
+  private static final String SYNTACTICALLY_INCORRECT_JSON = "{\"" + Person.USER_ID_FIELD_NAME + "\"";
+
+  private Person testPerson = Person.newBuilder()
+                                    .setEmail(EMAIL)
+                                    .setNickname(NICKNAME)
+                                    .setUserId(USER_ID)
+                                    .setPronouns(PRONOUNS)
+                                    .build();
+  private GetProfileServlet getProfileServlet;
+  private GetProfileServlet failingGetProfileServlet;
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper();
+  private StringWriter stringWriter = new StringWriter();
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+  @Mock private StorageHandlerApi successfulHandler;
+  @Mock private StorageHandlerApi failingHandler;
+  
+  @Before
+  public void setUp() throws IOException {
+    helper.setUp();
+
+    successfulHandler = mock(StorageHandlerApi.class);
+    when(successfulHandler.fetchPersonFromId(anyString())).thenReturn(testPerson);
+    getProfileServlet = new GetProfileServlet(successfulHandler);
+
+    failingHandler = mock(StorageHandlerApi.class);
+    when(failingHandler.fetchPersonFromId(anyString()))
+                       .thenThrow(new IllegalArgumentException(StorageHandler.PERSON_DOES_NOT_EXIST))
+                       .thenReturn(null);
```

Comment:
The `.thenReturn(null);` should be removed, as the `failingHandler` will throw an error beforehand. Super minor nit here but the `thenThrow` line also breaks the line length limit.

Reply:
Thanks for pointing this out! I wasn't entirely sure if I could remove the `.thenReturn(null)`, so I will fix that up now!